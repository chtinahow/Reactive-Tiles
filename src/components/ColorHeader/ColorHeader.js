import { registerHtml } from 'tram-one'
import './ColorHeader.scss'
import useColor from '../../hooks/useColor'

const html = registerHtml()

export default () => {
	const [color, incrementColor] = useColor()

	return html`
    <h1
      class="color-header"
      style="color:${color}"
      onclick=${incrementColor}
    >
      reactive-color
    </h1>
  `
}
