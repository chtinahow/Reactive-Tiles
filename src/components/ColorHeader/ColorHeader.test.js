import ColorHeader from './ColorHeader'

const mockColor = '#e6ef9f'
const mockUpdateColor = jest.fn()

// mock and spy on tram-one hooks
jest.mock('../../hooks/useColor', () => ({
	__esModule: true, // this property makes it work
	default: () => {
		return [mockColor, mockUpdateColor]
	}
}))

describe('ColorHeader', () => {
	it('should match snapshot', () => {
		const header = ColorHeader()
		expect(header.outerHTML).toMatchSnapshot()
	})

	it('should call mockSetColor on click', () => {
		const header = ColorHeader()
		header.click()
		expect(mockUpdateColor).toHaveBeenCalled()
	})
})
