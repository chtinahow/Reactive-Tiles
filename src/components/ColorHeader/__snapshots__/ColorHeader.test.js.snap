// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorHeader should match snapshot 1`] = `
"<h1 style=\\"color:#e6ef9f\\" class=\\"color-header\\">
      reactive-color
    </h1>"
`;
